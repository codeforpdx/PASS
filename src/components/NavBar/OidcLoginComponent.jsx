import React, { useState } from 'react';
import { TextField, Box, Button } from '@mui/material';
import { LoginButton } from '@inrupt/solid-ui-react';

// Utility Imports
import { SOLID_IDENTITY_PROVIDER } from '../../utils';

// Custom Hook Imports
import { useRedirectUrl } from '../../hooks';

const OidcLoginComponent = () => {
  const [oidcIssuer, setOidcIssuer] = useState(SOLID_IDENTITY_PROVIDER);
  const redirectUrl = useRedirectUrl();

  return (
    <>
      <Box sx={{ flexGrow: 1 }} />
      <TextField
        type="text"
        label="Pod Server URL"
        variant="filled"
        value={oidcIssuer}
        onChange={(e) => setOidcIssuer(e.target.value)}
        InputProps={{
          disableUnderline: true,
          'aria-label': 'OIDC Input Field'
        }}
        sx={{
          backgroundColor: 'white',
          borderRadius: '8px'
        }}
      />
      <Box sx={{ marginRight: '32px' }} />
      <LoginButton oidcIssuer={oidcIssuer} redirectUrl={redirectUrl}>
        <Button
          variant="contained"
          type="submit"
          color="secondary"
          size="large"
          aria-label="Login Button"
          onClick={() => {
            localStorage.setItem('oidcIssuer', oidcIssuer);
          }}
        >
          Login
        </Button>
      </LoginButton>
    </>
  );
};

export default OidcLoginComponent;
