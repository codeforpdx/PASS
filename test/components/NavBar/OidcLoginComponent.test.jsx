import { render, cleanup } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import { expect, it, vi, afterEach } from 'vitest';
import OidcLoginComponent from '../../../src/components/NavBar/OidcLoginComponent';

vi.mock('@inrupt/solid-ui-react', () => ({
  LoginButton: ({ children, oidcIssuer, redirectUrl }) => (
    <div>
      {oidcIssuer}
      {redirectUrl}
      {children}
    </div>
  )
}));

afterEach(() => {
  vi.clearAllMocks();
  localStorage.clear();
  cleanup();
});

vi.mock('../../../src/constants/', () => {
  const actual = vi.importActual('../../../src/constants/');
  return {
    ...actual,
    ENV: {
      VITE_SOLID_IDENTITY_PROVIDER: 'https://www.testurl.com/'
    }
  };
});

it('renders correctly', () => {
  const { container } = render(<OidcLoginComponent />);
  expect(container).toMatchSnapshot();
});

it('sets OIDC provider on login', async () => {
  const user = userEvent.setup();
  const { container, getByLabelText } = render(<OidcLoginComponent />);
  const input = getByLabelText('OIDC Input Field').querySelector('input');
  const login = getByLabelText('Login Button');
  vi.spyOn(Storage.prototype, 'setItem');
  await user.clear(input);
  await user.type(input, 'oidc.provider.url');
  expect(input.value).toBe('oidc.provider.url');
  await user.click(login);
  expect(localStorage.setItem).toBeCalledWith('oidcIssuer', 'oidc.provider.url');
  expect(container).toMatchSnapshot();
});
