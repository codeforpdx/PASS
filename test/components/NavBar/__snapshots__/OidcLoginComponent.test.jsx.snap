// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly 1`] = `
<div>
  <div
    class="MuiBox-root css-i9gxme"
  />
  <div
    class="MuiFormControl-root MuiTextField-root css-1q1ral2-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled css-e4w4as-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="false"
      for=":r0:"
      id=":r0:-label"
    >
      Pod Server URL
    </label>
    <div
      aria-label="OIDC Input Field"
      class="MuiInputBase-root MuiFilledInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1pw81iq-MuiInputBase-root-MuiFilledInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiFilledInput-input css-10botns-MuiInputBase-input-MuiFilledInput-input"
        id=":r0:"
        type="text"
        value="https://www.testurl.com/"
      />
    </div>
  </div>
  <div
    class="MuiBox-root css-120gsus"
  />
  <div>
    https://www.testurl.com/
    <button
      aria-label="Login Button"
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeLarge MuiButton-containedSizeLarge css-1xnmwiw-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="submit"
    >
      Login
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`sets OIDC provider on login 1`] = `
<div>
  <div
    class="MuiBox-root css-i9gxme"
  />
  <div
    class="MuiFormControl-root MuiTextField-root css-1q1ral2-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled css-o943dk-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for=":r1:"
      id=":r1:-label"
    >
      Pod Server URL
    </label>
    <div
      aria-label="OIDC Input Field"
      class="MuiInputBase-root MuiFilledInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1pw81iq-MuiInputBase-root-MuiFilledInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiFilledInput-input css-10botns-MuiInputBase-input-MuiFilledInput-input"
        id=":r1:"
        type="text"
        value="oidc.provider.url"
      />
    </div>
  </div>
  <div
    class="MuiBox-root css-120gsus"
  />
  <div>
    oidc.provider.url
    <button
      aria-label="Login Button"
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeLarge MuiButton-containedSizeLarge css-1xnmwiw-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="submit"
    >
      Login
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      >
        <span
          class="css-y4cjyz-MuiTouchRipple-ripple MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
          style="width: 2.8284271247461903px; height: 2.8284271247461903px; top: -1.4142135623730951px; left: -1.4142135623730951px;"
        >
          <span
            class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
          />
        </span>
      </span>
    </button>
  </div>
</div>
`;
